scripts/build-containers.sh

```diff
@@ -1,7 +1,22 @@
 #!/bin/sh
 set -eux
 
-time docker build . -t local/substrate-relay --build-arg=PROJECT=substrate-relay
-time docker build . -t local/rialto-bridge-node --build-arg=PROJECT=rialto-bridge-node
-time docker build . -t local/millau-bridge-node --build-arg=PROJECT=millau-bridge-node
-time docker build . -t local/rialto-parachain-collator --build-arg=PROJECT=rialto-parachain-collator
+if [ -z "${LOCAL:-}" ]; then
+    time docker build . -t local/substrate-relay --build-arg=PROJECT=substrate-relay
+    time docker build . -t local/rialto-bridge-node --build-arg=PROJECT=rialto-bridge-node
+    time docker build . -t local/millau-bridge-node --build-arg=PROJECT=millau-bridge-node
+    time docker build . -t local/rialto-parachain-collator --build-arg=PROJECT=rialto-parachain-collator
+else
+    if [ -z "${SKIP_BUILD:-}" ]; then
+        time cargo build -p substrate-relay --release
+        time cargo build -p rialto-bridge-node --release
+        time cargo build -p millau-bridge-node --release
+        time cargo build -p rialto-parachain-collator --release
+    fi
+
+    # following (using DOCKER_BUILDKIT) requires docker 19.03 or above
+    DOCKER_BUILDKIT=1 time docker build . -f local.Dockerfile -t local/substrate-relay --build-arg=PROJECT=substrate-relay
+    DOCKER_BUILDKIT=1 time docker build . -f local.Dockerfile -t local/rialto-bridge-node --build-arg=PROJECT=rialto-bridge-node
+    DOCKER_BUILDKIT=1 time docker build . -f local.Dockerfile -t local/millau-bridge-node --build-arg=PROJECT=millau-bridge-node
+    DOCKER_BUILDKIT=1 time docker build . -f local.Dockerfile -t local/rialto-parachain-collator --build-arg=PROJECT=rialto-parachain-collator
+fi
```

Comment:
Nit: Could we also clean the old images at the end ?

Reply:
I'm not sure what old images do you mean?

Reply:
For example when we generate a new `local/rialto-bridge-node:latest`, AFAIK the previous `local/rialto-bridge-node:latest` will not be removed automatically. Or at least there will be layers of its storage that are not needed anymore, but they are kept. It would be nice to remove them. Not sure if possible. I usually run `docker rmi -f $(docker images -f dangling=true -q)`, but this deletes all dangling images.

Reply:
OK, thanks for explanation. But imo let's not do this in script - removal looks like a manual action to me. And `docker system prune` (or something like that) isn't super complex to call sometimes.

Reply:
Ok, sure